### ISSUE TITLE
Can't connect to the room - D/Room: onConnectFailure()

### ISSUE THREAD

**Issue by vboko** [twilio/video-quickstart-android#302]
### Description

I did every step described in instructions for video-quickstart-android.
I am not able to connect to the room.

**D/Room: onConnectFailure()**

### Steps to Reproduce

1. Check out the code for video-quickstart-android
2. Add the access token string copied from the console to a variable named TWILIO_ACCESS_TOKEN in your local.properties file
3. Run quickstart app
4. Enter room name and connect to it

#### Expected Behavior

Successful connection to the room

#### Actual Behavior

Connection failed, video chat is not established

#### Reproduces how Often

100%

#### Logs

05-24 18:06:45.980 17117-18357/com.twilio.video.quickstart D/TwilioVideo: [Platform]:onMessage
    [Platform]:connectOnNotifier
05-24 18:06:45.983 17117-18357/com.twilio.video.quickstart D/TwilioVideo: [Platform]:AndroidRoomObserver
05-24 18:06:45.998 17117-18357/com.twilio.video.quickstart D/TwilioVideo: [Platform]:Create PlatformInfo
05-24 18:06:45.998 17117-18357/com.twilio.video.quickstart I/TwilioVideo: [Core]:Using user provider notifier queue
05-24 18:06:46.003 17117-18357/com.twilio.video.quickstart I/TwilioVideo: [Core]:Creating SIP stack ...
05-24 18:06:46.004 17117-18357/com.twilio.video.quickstart I/TwilioVideo: [Core]:Adding IPv4/IPv6 TLS transports ...
    [Signaling]:RESIP::TRANSPORT: Connection::Connection: new connection created to who: [ V4 0.0.0.0:0 UNKNOWN_TRANSPORT target domain=unspecified mFlowKey=0 ]
    [Signaling]:RESIP::TRANSPORT: Creating TLS transport for domain  interface= port=0
05-24 18:06:46.005 17117-18357/com.twilio.video.quickstart I/TwilioVideo: [Signaling]:RESIP::TRANSPORT: Connection::Connection: new connection created to who: [ V4 0.0.0.0:0 UNKNOWN_TRANSPORT target domain=unspecified mFlowKey=0 ]
    [Signaling]:RESIP::TRANSPORT: Creating TLS transport for domain  interface= port=0
05-24 18:06:46.006 17117-18357/com.twilio.video.quickstart D/TwilioVideo: [Core]:RoomSignalingImpl::RoomSignalingImpl()
05-24 18:06:46.006 17117-18357/com.twilio.video.quickstart I/TwilioVideo: [Core]:Starting SIP signaling stack worker thread ...
05-24 18:06:46.009 17117-18361/com.twilio.video.quickstart D/TwilioVideo: [Core]:Starting StackThread runloop.
05-24 18:06:46.010 17117-18361/com.twilio.video.quickstart V/TwilioVideo: [Core]:Select for 2147483.647 seconds.
    [Core]:Received an application message.
05-24 18:06:46.010 17117-18361/com.twilio.video.quickstart I/TwilioVideo: [Core]:Querying SIP edge (mobile-endpoint.twilio.com) IP addresses ...
    [Core]:Looking up: mobile-endpoint.twilio.com
05-24 18:06:46.031 17117-17117/com.twilio.video.quickstart D/AudioManager: dispatching onAudioFocusChange(-2) to android.media.AudioManager@44e294ecom.twilio.video.quickstart.activity.VideoActivity$11@6318a6f
05-24 18:06:46.037 17117-17117/com.twilio.video.quickstart D/Video: Ignoring network event, sticky broadcast
05-24 18:06:46.132 17117-18361/com.twilio.video.quickstart D/TwilioVideo: [Core]:Address family: 2: 34.203.250.168
    [Core]:Address family: 2: 34.203.250.168
05-24 18:06:46.132 17117-18361/com.twilio.video.quickstart I/TwilioVideo: [Core]:Setting SIP edge (outbound proxy) to: sips:34.203.250.168:443
    [Core]:DNS resolution was successful
05-24 18:06:46.133 17117-18361/com.twilio.video.quickstart D/TwilioVideo: [Core]:Sending a ping message ...
05-24 18:06:46.137 17117-18361/com.twilio.video.quickstart I/TwilioVideo: [Signaling]:RESIP::TRANSPORT: Connection::Connection: new connection created to who: [ V4 34.203.250.168:443 TLS target domain=unspecified mFlowKey=86 ]
    [Signaling]:RESIP::TRANSPORT: Creating TLS connection for domain  [ V4 34.203.250.168:443 TLS target domain=unspecified mFlowKey=0 ] on 86
05-24 18:06:46.137 17117-18361/com.twilio.video.quickstart V/TwilioVideo: [Core]:Select for 29.995 seconds.
    [Core]:Select for 29.995 seconds.
05-24 18:06:46.181 17117-17122/com.twilio.video.quickstart I/zygote64: Do partial code cache collection, code=249KB, data=181KB
05-24 18:06:46.182 17117-17122/com.twilio.video.quickstart I/zygote64: After code cache collection, code=249KB, data=181KB
    Increasing code cache capacity to 1024KB
05-24 18:06:47.354 17117-18360/com.twilio.video.quickstart V/TwilioVideo: [Core]:Response: OK, code: 200
05-24 18:06:47.355 17117-18360/com.twilio.video.quickstart D/TwilioVideo: [Core]:Parsing 'video' endpoint config: {"video":{"network_traversal_service":{"ttl":3600,"date_created":"Thu, 24 May 2018 15:06:47 +0000","date_updated":"Thu, 24 May 2018 15:06:47 +0000","ice_servers":[{"username":"4a82f7b8ed21bd3a2b9984e28c9380ee049637d233826d25219a4d0df53d9aad","credential":"CpFUVQFwOuJ9T1Q3Xs+FzOBw3l/9YJrh2oVY7Mvg0s4=","urls":"turn:global.turn.twilio.com:3478?transport=udp"},{"username":"4a82f7b8ed21bd3a2b9984e28c9380ee049637d233826d25219a4d0df53d9aad","credential":"CpFUVQFwOuJ9T1Q3Xs+FzOBw3l/9YJrh2oVY7Mvg0s4=","urls":"turns:global.turn.twilio.com:443?transport=tcp"}]}}}
05-24 18:06:47.355 17117-18359/com.twilio.video.quickstart D/TwilioVideo: [Core]:RoomSignalingImpl: State transition successful: kInit -> kConnecting
05-24 18:06:47.355 17117-18359/com.twilio.video.quickstart I/TwilioVideo: [Core]:Creating peer connection ...
05-24 18:06:47.361 17117-18359/com.twilio.video.quickstart I/TwilioVideo: [Core]:Adding local stream to peer connection ...
05-24 18:06:47.361 17117-18359/com.twilio.video.quickstart D/TwilioVideo: [Core]:Open -> Updating. Process an event
    [Core]:Create local offer: 1c7d5513f6cc1e5CfC3a7b09bdaD6dbC
05-24 18:06:47.366 17117-17240/com.twilio.video.quickstart D/TwilioVideo: [Core]:onCreateSessionLocalDescription 1c7d5513f6cc1e5CfC3a7b09bdaD6dbC
05-24 18:06:47.368 17117-17240/com.twilio.video.quickstart D/TwilioVideo: [Core]:SDP Utils: New SSRC group 985458160 2169018910 --- Original SSRC group 985458160 2169018910
05-24 18:06:47.372 17117-17240/com.twilio.video.quickstart D/TwilioVideo: [Core]:PeerConnection state: Updating -> Waiting
05-24 18:06:47.372 17117-18359/com.twilio.video.quickstart I/TwilioVideo: [Core]:Local SDP is ready
05-24 18:06:47.373 17117-18359/com.twilio.video.quickstart I/TwilioVideo: [Core]:Queue Description: 1 for PeerConnection: 1c7d5513f6cc1e5CfC3a7b09bdaD6dbC.
05-24 18:06:47.373 17117-18361/com.twilio.video.quickstart V/TwilioVideo: [Core]:Received an application message.
05-24 18:06:47.374 17117-18361/com.twilio.video.quickstart D/TwilioVideo: [Core]:Adding X-Twilio-AccessToken header.
    [Core]:
    Sending outgoing SIP message
    INVITE sip:mobile-endpoint.twilio.com;transport=tls SIP/2.0
    Via: SIP/2.0/TLS 127.0.0.1;branch=z9hG4bK-524287-1---a69c168833ceff14;rport
    Max-Forwards: 70
    Contact: <sip:dDcb6BB43dB22a5207eA92Aec8bEc9eB@127.0.0.1;transport=tls;ob>
    To: <sip:orchestrator@mobile-endpoint.twilio.com>
    From: "dDcb6BB43dB22a5207eA92Aec8bEc9eB"<sip:dDcb6BB43dB22a5207eA92Aec8bEc9eB@mobile-endpoint.twilio.com>;tag=7c67402b
    Call-ID: eunaAbAvn5TPEhwpQaZimw..
    CSeq: 1 INVITE
    Session-Expires: 120
    Allow: INVITE, ACK, CANCEL, OPTIONS, BYE, UPDATE, NOTIFY
    Content-Type: application/room-signaling+json
    Supported: timer, outbound, path, gruu, room-signaling
    User-Agent: TwilioVideo SDK
    X-Twilio-AccessToken: eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiIsImN0eSI6InR3aWxpby1mcGE7dj0xIn0.eyJqdGkiOiJTSzgzOTRmNmRhMmY2YmNlYTRhNDcyZDEwMGQzY2E4YWJjLTE1MjcxNzM3NzkiLCJpc3MiOiJTSzgzOTRmNmRhMmY2YmNlYTRhNDcyZDEwMGQzY2E4YWJjIiwic3ViIjoiQUM4OWE0NGQ1ZTdiYzA3ZGU3ZWRiZDQ4NWZjNjUyYTdlYSIsImV4cCI6MTUyNzE3NzM3OSwiZ3JhbnRzIjp7ImlkZW
05-24 18:06:47.375 17117-18361/com.twilio.video.quickstart W/TwilioVideo: [Signaling]:RESIP::TRANSPORT: Can't find matching transport [ V4 127.0.0.1:0 TLS target domain=unspecified mFlowKey=0 ]
05-24 18:06:47.377 17117-18361/com.twilio.video.quickstart V/TwilioVideo: [Core]:Select for 28.756 seconds.
    [Core]:Select for 28.756 seconds.
05-24 18:06:48.410 17117-17246/com.twilio.video.quickstart I/org.webrtc.Logging: WebRtcAudioManager: VOICE_CALL stream volume: 7 (max=7)
05-24 18:06:58.410 17117-17246/com.twilio.video.quickstart I/org.webrtc.Logging: WebRtcAudioManager: VOICE_CALL stream volume: 7 (max=7)
05-24 18:07:00.372 17117-17126/com.twilio.video.quickstart E/SpellCheckerSession: ignoring processOrEnqueueTask due to unexpected mState=TASK_CLOSE scp.mWhat=TASK_CLOSE
05-24 18:07:08.411 17117-17246/com.twilio.video.quickstart I/org.webrtc.Logging: WebRtcAudioManager: VOICE_CALL stream volume: 7 (max=7)
05-24 18:07:16.146 17117-18361/com.twilio.video.quickstart D/TwilioVideo: [Core]:Sending a ping message ...
05-24 18:07:16.148 17117-18361/com.twilio.video.quickstart V/TwilioVideo: [Core]:Select for 3.226 seconds.
    [Core]:Select for 3.226 seconds.
05-24 18:07:18.411 17117-17246/com.twilio.video.quickstart I/org.webrtc.Logging: WebRtcAudioManager: VOICE_CALL stream volume: 7 (max=7)
05-24 18:07:19.377 17117-18361/com.twilio.video.quickstart D/TwilioVideo: [Core]:
    Resiprocate generated the SIP message
    SIP/2.0 408 Request Timeout
    Via: SIP/2.0/TLS 127.0.0.1;branch=z9hG4bK-524287-1---a69c168833ceff14;rport
    To: <sip:orchestrator@mobile-endpoint.twilio.com>;tag=ef87149f
    From: "dDcb6BB43dB22a5207eA92Aec8bEc9eB"<sip:dDcb6BB43dB22a5207eA92Aec8bEc9eB@mobile-endpoint.twilio.com>;tag=7c67402b
    Call-ID: eunaAbAvn5TPEhwpQaZimw..
    CSeq: 1 INVITE
    Content-Length: 0
05-24 18:07:19.378 17117-18361/com.twilio.video.quickstart D/TwilioVideo: [Core]:Process INVITE response with code 408
05-24 18:07:19.378 17117-18361/com.twilio.video.quickstart V/TwilioVideo: [Core]:Received error message in connecting state.
05-24 18:07:19.378 17117-18361/com.twilio.video.quickstart D/TwilioVideo: [Core]:onTerminated: code: 53000 msg: Signaling connection error explanation: SIP error 408
    [Core]:RoomSignalingImpl: State transition successful: kConnecting -> kDisconnecting
05-24 18:07:19.378 17117-18361/com.twilio.video.quickstart V/TwilioVideo: [Core]:Select for 26.768 seconds.
05-24 18:07:19.378 17117-18359/com.twilio.video.quickstart D/TwilioVideo: [Core]:Disconnecting
    [Core]:Canceling reconnect retry timer.
05-24 18:07:19.378 17117-18359/com.twilio.video.quickstart I/TwilioVideo: [Core]:Close PeerConnectionSignaling's underlying PeerConnection
05-24 18:07:19.380 17117-17240/com.twilio.video.quickstart D/TwilioVideo: [Core]:Discarding ice connection state update because our state is closed.
05-24 18:07:19.380 17117-17240/com.twilio.video.quickstart I/TwilioVideo: [Core]:Ice Gathering Complete.
05-24 18:07:19.380 17117-17240/com.twilio.video.quickstart D/TwilioVideo: [Core]:Discarding ice candidate update because our state is closed.
05-24 18:07:19.384 17117-18359/com.twilio.video.quickstart I/TwilioVideo: [Core]:Done closing the PeerConnection
05-24 18:07:19.385 17117-18359/com.twilio.video.quickstart I/TwilioVideo: [Core]:PeerConnectionSignaling destroyed
05-24 18:07:19.385 17117-18359/com.twilio.video.quickstart D/TwilioVideo: [Core]:Canceling disconnect timer.
    [Core]:RoomSignalingImpl: State transition successful: kDisconnecting -> kDisconnected
05-24 18:07:19.385 17117-18359/com.twilio.video.quickstart I/TwilioVideo: [Core]:Shutdown and join the signaling stack's thread.
05-24 18:07:19.385 17117-18361/com.twilio.video.quickstart D/TwilioVideo: [Core]:Shutting down StackThread runloop.
05-24 18:07:19.391 17117-18359/com.twilio.video.quickstart V/TwilioVideo: [Core]:virtual twilio::signaling::InterruptableStackThread::~InterruptableStackThread()
05-24 18:07:19.393 17117-18357/com.twilio.video.quickstart I/TwilioVideo: [Core]:Failed to connect to a Room with name: test
05-24 18:07:19.393 17117-18357/com.twilio.video.quickstart D/TwilioVideo: [Platform]:onConnectFailure
05-24 18:07:19.398 17117-18357/com.twilio.video.quickstart D/TwilioVideo: [Platform]:Java_com_twilio_video_Room_nativeReleaseRoom
    [Platform]:release
    [Platform]:~AndroidStatsObserver
05-24 18:07:19.399 17117-18357/com.twilio.video.quickstart D/TwilioVideo: [Platform]:~AndroidRoomObserver
    [Core]:RoomSignalingImpl::~RoomSignalingImpl()
    [Core]:RoomSignalingImpl: State Transition Failure: kDisconnected -> kDisconnecting
05-24 18:07:19.400 17117-17117/com.twilio.video.quickstart D/Room: onConnectFailure()
05-24 18:07:19.400 17117-17117/com.twilio.video.quickstart D/TwilioVideo: [Platform]:Java_com_twilio_video_Room_nativeRelease
    [Platform]:~RoomDelegate
05-24 18:07:19.449 17117-17117/com.twilio.video.quickstart D/TwilioVideo: [Platform]:Free PlatformInfo

#### Video Android SDK

2.0.2

#### Android API

27

#### Android Device

Nexus 5x

**Comment by aaalaniz** [comment]
Hey @vboko 

I noticed this from the log

```
SIP/2.0 408 Request Timeout
Via: SIP/2.0/TLS 127.0.0.1;branch=z9hG4bK-524287-1---a69c168833ceff14;rport
To: sip:orchestrator@mobile-endpoint.twilio.com;tag=ef87149f
From: "dDcb6BB43dB22a5207eA92Aec8bEc9eB"sip:dDcb6BB43dB22a5207eA92Aec8bEc9eB@mobile-endpoint.twilio.com;tag=7c67402b
Call-ID: eunaAbAvn5TPEhwpQaZimw..
CSeq: 1 INVITE
Content-Length: 0
05-24 18:07:19.378 17117-18361/com.twilio.video.quickstart D/TwilioVideo: [Core]:Process INVITE response with code 408
05-24 18:07:19.378 17117-18361/com.twilio.video.quickstart V/TwilioVideo: [Core]:Received error message in connecting state.
05-24 18:07:19.378 17117-18361/com.twilio.video.quickstart D/TwilioVideo: [Core]:onTerminated: code: 53000 msg: Signaling connection error explanation: SIP error 408
```

Does the network you are connecting to a Room on have any restrictions? Are you seeing this problem with any of our other SDK quickstarts (JS or iOS) on that network.

Thanks!

**Comment by vboko** [comment]
@aaalaniz I don't think that the network has restrictions, I launched old examples few months and they worked fine. I didn't modify anything on my router.
I tried to use my home network with an emulator and real device as well as cellular network.
The result is the same, can't connect to the room.

Are there any other options I have to try to resolve it?

PS: I will try to run iOS example and let you know

**Comment by aaalaniz** [comment]
Hey @vboko 

Are you still encountering this issue? What did you find when trying the other platform quickstarts?

Thanks!

**Comment by vboko** [comment]
@aaalaniz I wasn't able to try out samples for other platforms, but I found out one thing when my Slack went down and couldn't connect yesterday. They responded to me that it might be because of IP addresses blocking in Russia (https://www.reuters.com/article/russia-telegram-ipaddresses/russia-tells-amazon-google-their-ip-addresses-blocked-because-of-telegram-ifax-idUSR4N1RQ00P)
Probably this blocking affected Twilio servers as well. So I tried to use VPN.
After that example worked fine. But when I disable VPN, I encounter that issue.
I think it can be closed now.
